rust/noosphere-collections/src/hamt/node.rs

```diff
@@ -2,6 +2,7 @@
 // Source copyright and license:
 // Copyright 2019-2022 ChainSafe Systems
 // SPDX-License-Identifier: Apache-2.0, MIT
+#![allow(clippy::only_used_in_recursion)]
```

Comment:
There's a warning w/r/t `depth` argument in the below functions which isn't currently used. We could remove it, though there may be a future need for it, deferring to you @cdata